src/Microsoft.AspNetCore.SignalR.Client.Core/HubConnection.cs

```diff
@@ -159,179 +109,283 @@ public IDisposable On(string methodName, Type[] parameterTypes, Func<object[], o
             return new Subscription(invocationHandler, invocationList);
         }
 
-        public async Task<ChannelReader<object>> StreamAsChannelAsync(string methodName, Type returnType, object[] args, CancellationToken cancellationToken = default)
+        public async Task<ChannelReader<object>> StreamAsChannelAsync(string methodName, Type returnType, object[] args, CancellationToken cancellationToken = default) =>
+            await StreamAsChannelAsyncCore(methodName, returnType, args, cancellationToken).ForceAsync();
+
+        public async Task<object> InvokeAsync(string methodName, Type returnType, object[] args, CancellationToken cancellationToken = default) =>
+            await InvokeAsyncCore(methodName, returnType, args, cancellationToken).ForceAsync();
+
+        // REVIEW: We don't generally use cancellation tokens when writing to a pipe because the asynchrony is only the result of backpressure.
+        // However, this would be the only "invocation" method _without_ a cancellation token... which is odd.
+        public async Task SendAsync(string methodName, object[] args, CancellationToken cancellationToken = default) =>
+            await SendAsyncCore(methodName, args).ForceAsync();
+
+        private async Task StartAsyncCore()
         {
-            return await StreamAsChannelAsyncCore(methodName, returnType, args, cancellationToken).ForceAsync();
+            await _startLock.WaitAsync();
+            try
+            {
+                if (_connection != null)
+                {
+                    // We're already connected
+                    return;
+                }
+
+                CheckDisposed();
+
+                Log.Starting(_logger);
+
+                _connectionActive = new CancellationTokenSource();
```

Comment:
It looks like this is used as a glorified bool. Where is it passed in?

Reply:
It was a CTS before. Now that we lock more, it can probably be a `bool`